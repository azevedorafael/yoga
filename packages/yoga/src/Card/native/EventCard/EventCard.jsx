import React from 'react';
import styled from 'styled-components';
import { oneOf, shape, string } from 'prop-types';
import { View } from 'react-native';

import Card from '../Card';
import Text from '../../../Text';

const Event = styled(Card)`
  flex-direction: row;

  width: 280px;
  height: 104px;
  padding: 0;
`;

const DateInfo = styled(View)`
  ${({
    variant,
    theme: {
      yoga: {
        components: {
          card: { event },
        },
        colors: { [variant]: color = {} },
      },
    },
  }) => `
    justify-content: center;
    align-items: center;

    width: 56px;

    border-top-left-radius: ${event.date.radius}px;
    border-bottom-left-radius: ${event.date.radius}px;
    background-color: ${color[2]};
  `}
`;

const EventInfo = styled(View)`
  ${({
    theme: {
      yoga: {
        components: {
          card: { event },
        },
      },
    },
  }) => `
    padding: ${event.info.padding}px;
  `}
`;

const DayOfWeek = styled(Text.Small)`
  ${({
    theme: {
      yoga: {
        components: {
          card: { event },
        },
      },
    },
  }) => `
    padding-bottom: ${event.date.dayOfWeek.padding.bottom}px;

    font-weight: ${event.date.dayOfWeek.fontWeight};
  `}
`;

const Month = styled(Text.Small)`
  ${({
    theme: {
      yoga: {
        components: {
          card: { event },
        },
      },
    },
  }) => `
    font-weight: ${event.date.month.fontWeight};
  `}
`;

const Name = styled(Text.H4)`
  ${({
    theme: {
      yoga: {
        components: {
          card: { event },
        },
      },
    },
  }) => `
    padding-bottom: ${event.info.name.padding.bottom}px;

    font-weight: ${event.info.name.fontWeight};
  `}
`;

const Place = styled(Text.Small)`
  ${({
    theme: {
      yoga: {
        components: {
          card: { event },
        },
      },
    },
  }) => `
    padding-bottom: ${event.info.place.padding.bottom}px;

    color: ${event.info.place.color};
  `}
`;

const EventCard = ({ event, date, variant, ...rest }) => (
  <Event {...rest}>
    <DateInfo variant={variant}>
      <DayOfWeek inverted>{date.weekday}</DayOfWeek>
      <Text.H3 inverted>{date.day}</Text.H3>
      <Month inverted>{date.month}</Month>
    </DateInfo>
    <EventInfo>
      <Name>{event.name}</Name>
      <Place>{event.place}</Place>
      <Text.Tiny>{event.time}</Text.Tiny>
    </EventInfo>
  </Event>
);

EventCard.propTypes = {
  event: shape({
    name: string,
    place: string,
    time: string,
  }).isRequired,
  date: shape({
    day: string,
    weekday: string,
    month: string,
  }).isRequired,
  /** style the card following the theme (primary, secondary, tertiary) */
  variant: oneOf(['primary', 'secondary', 'tertiary']),
};

EventCard.defaultProps = {
  variant: 'secondary',
};

EventCard.displayName = 'EventCard';

export default EventCard;
