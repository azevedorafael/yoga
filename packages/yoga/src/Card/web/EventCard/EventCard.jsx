import React from 'react';
import styled, { withTheme, css } from 'styled-components';
import { oneOf, shape, string } from 'prop-types';
import { Clock } from '@gympass/yoga-icons';

import Card from '../Card';
import Text from '../../../Text';

const Event = styled(Card)`
  display: flex;
  flex-direction: row;

  width: 100%;
  padding: 0;
`;

const EventInfo = styled.div`
  ${({
    theme: {
      yoga: {
        components: {
          card: { event },
        },
      },
    },
  }) => css`
    padding: ${event.info.padding}px;

    ${Text.H4} {
      padding-bottom: ${event.info.name.padding.bottom}px;

      font-weight: ${event.info.name.fontWeight};
    }

    ${Text.Small} {
      padding-bottom: ${event.info.place.padding.bottom}px;

      color: ${event.info.place.color};
    }
  `}
`;

const EventTime = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
`;

const DateInfo = styled.div`
  ${({
    variant,
    theme: {
      yoga: {
        components: {
          card: { event },
        },
        colors: { [variant]: color },
      },
    },
  }) => `
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    flex-shrink: 0;

    width: 56px;

    border-top-left-radius: ${event.date.radius}px;
    border-bottom-left-radius: ${event.date.radius}px;
    background-color: ${color[2]};
  `}
`;

const Weekday = styled(Text.Small)`
  ${({
    theme: {
      yoga: {
        components: {
          card: { event },
        },
      },
    },
  }) => `
    padding-bottom: ${event.date.weekday.padding.bottom}px;

    font-weight: ${event.date.weekday.fontWeight};
  `}
`;

const Month = styled(Text.Small)`
  ${({
    theme: {
      yoga: {
        components: {
          card: { event },
        },
      },
    },
  }) => `
    font-weight: ${event.date.month.fontWeight};
  `}
`;

const EventCard = ({
  event,
  date,
  variant,
  theme: {
    yoga: {
      components: {
        card: {
          event: { icon },
        },
      },
    },
  },
  ...rest
}) => (
  <Event {...rest}>
    <DateInfo variant={variant}>
      <Weekday inverted>{date.weekday}</Weekday>
      <Text.H3 inverted>{date.day}</Text.H3>
      <Month inverted>{date.month}</Month>
    </DateInfo>
    <EventInfo>
      <Text.H4>{event.name}</Text.H4>
      <Text.Small>{event.place}</Text.Small>
      <EventTime>
        <Clock fill={icon.fill} style={{ marginRight: 5 }} />
        <Text.Tiny>{event.time}</Text.Tiny>
      </EventTime>
    </EventInfo>
  </Event>
);

EventCard.propTypes = {
  /** event information: { name (string), place (string), time (string) } */
  event: shape({
    name: string,
    place: string,
    time: string,
  }).isRequired,
  /** date information: { day (string), weekday (string), month (string) } */
  date: shape({
    day: string,
    weekday: string,
    month: string,
  }).isRequired,
  /** style the card following the theme (primary, secondary, tertiary) */
  variant: oneOf(['primary', 'secondary', 'tertiary']),
};

EventCard.defaultProps = {
  variant: 'secondary',
};

EventCard.displayName = 'EventCard';

export default withTheme(EventCard);
