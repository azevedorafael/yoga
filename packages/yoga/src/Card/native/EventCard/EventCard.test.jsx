import React from 'react';
import { render } from '@testing-library/react-native';
import { ThemeProvider } from '../../..';
import EventCard from '.';

const event = {
  name: 'Yoga Class',
  place: 'Gympass',
  day: '19',
  dayOfWeek: 'thu',
  month: 'dec',
  time: '19 am',
};

describe('<Card />', () => {
  describe('Snapshots', () => {
    it('should match snapshot with default EventCard', () => {
      const { container } = render(
        <ThemeProvider>
          <EventCard event={event} />
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });
    it('should match snapshot with variant primary EventCard', () => {
      const { container } = render(
        <ThemeProvider>
          <EventCard event={event} variant="primary" />
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });
    it('should match snapshot with variant secondary EventCard', () => {
      const { container } = render(
        <ThemeProvider>
          <EventCard event={event} variant="secondary" />
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });
    it('should match snapshot with variant tertiary EventCard', () => {
      const { container } = render(
        <ThemeProvider>
          <EventCard event={event} variant="tertiary" />
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });
  });
});
