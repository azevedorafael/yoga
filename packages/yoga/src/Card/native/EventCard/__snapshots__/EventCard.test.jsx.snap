// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Snapshots should match snapshot with default EventCard 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 8,
          "elevation": 4,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "shadowColor": "rgba(152, 152, 166, 0.25)",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 6,
        },
        Object {
          "flexDirection": "row",
          "height": 104,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "width": 280,
        },
      ]
    }
    variant=""
    variantIntensity={3}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#8051E1",
            "borderBottomLeftRadius": 8,
            "borderTopLeftRadius": 8,
            "justifyContent": "center",
            "width": 56,
          },
        ]
      }
      variant="secondary"
    >
      <Text
        inverted={true}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Open Sans",
              "fontSize": 14,
              "fontWeight": "600",
              "lineHeight": 20,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 8,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
      >
        thu
      </Text>
      <Text
        inverted={true}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Open Sans",
              "fontSize": 20,
              "fontWeight": "700",
              "lineHeight": 24,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
      >
        19
      </Text>
      <Text
        inverted={true}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Open Sans",
              "fontSize": 14,
              "fontWeight": "600",
              "lineHeight": 20,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
      >
        dec
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#41414A",
              "fontFamily": "Open Sans",
              "fontSize": 16,
              "fontWeight": "600",
              "lineHeight": 20,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 8,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
      >
        Yoga Class
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#6B6B78",
              "fontFamily": "Open Sans",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 8,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
      >
        Gympass
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#41414A",
              "fontFamily": "Open Sans",
              "fontSize": 12,
              "fontWeight": "400",
              "lineHeight": 16,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
      >
        19 am
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Card /> Snapshots should match snapshot with variant primary EventCard 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 8,
          "elevation": 4,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "shadowColor": "rgba(152, 152, 166, 0.25)",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 6,
        },
        Object {
          "flexDirection": "row",
          "height": 104,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "width": 280,
        },
      ]
    }
    variant=""
    variantIntensity={3}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#F78D82",
            "borderBottomLeftRadius": 8,
            "borderTopLeftRadius": 8,
            "justifyContent": "center",
            "width": 56,
          },
        ]
      }
      variant="primary"
    >
      <Text
        inverted={true}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Open Sans",
              "fontSize": 14,
              "fontWeight": "600",
              "lineHeight": 20,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 8,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
      >
        thu
      </Text>
      <Text
        inverted={true}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Open Sans",
              "fontSize": 20,
              "fontWeight": "700",
              "lineHeight": 24,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
      >
        19
      </Text>
      <Text
        inverted={true}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Open Sans",
              "fontSize": 14,
              "fontWeight": "600",
              "lineHeight": 20,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
      >
        dec
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#41414A",
              "fontFamily": "Open Sans",
              "fontSize": 16,
              "fontWeight": "600",
              "lineHeight": 20,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 8,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
      >
        Yoga Class
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#6B6B78",
              "fontFamily": "Open Sans",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 8,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
      >
        Gympass
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#41414A",
              "fontFamily": "Open Sans",
              "fontSize": 12,
              "fontWeight": "400",
              "lineHeight": 16,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
      >
        19 am
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Card /> Snapshots should match snapshot with variant secondary EventCard 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 8,
          "elevation": 4,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "shadowColor": "rgba(152, 152, 166, 0.25)",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 6,
        },
        Object {
          "flexDirection": "row",
          "height": 104,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "width": 280,
        },
      ]
    }
    variant=""
    variantIntensity={3}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#8051E1",
            "borderBottomLeftRadius": 8,
            "borderTopLeftRadius": 8,
            "justifyContent": "center",
            "width": 56,
          },
        ]
      }
      variant="secondary"
    >
      <Text
        inverted={true}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Open Sans",
              "fontSize": 14,
              "fontWeight": "600",
              "lineHeight": 20,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 8,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
      >
        thu
      </Text>
      <Text
        inverted={true}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Open Sans",
              "fontSize": 20,
              "fontWeight": "700",
              "lineHeight": 24,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
      >
        19
      </Text>
      <Text
        inverted={true}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Open Sans",
              "fontSize": 14,
              "fontWeight": "600",
              "lineHeight": 20,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
      >
        dec
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#41414A",
              "fontFamily": "Open Sans",
              "fontSize": 16,
              "fontWeight": "600",
              "lineHeight": 20,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 8,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
      >
        Yoga Class
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#6B6B78",
              "fontFamily": "Open Sans",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 8,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
      >
        Gympass
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#41414A",
              "fontFamily": "Open Sans",
              "fontSize": 12,
              "fontWeight": "400",
              "lineHeight": 16,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
      >
        19 am
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Card /> Snapshots should match snapshot with variant tertiary EventCard 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 8,
          "elevation": 4,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "shadowColor": "rgba(152, 152, 166, 0.25)",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 6,
        },
        Object {
          "flexDirection": "row",
          "height": 104,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "width": 280,
        },
      ]
    }
    variant=""
    variantIntensity={3}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#3FA99B",
            "borderBottomLeftRadius": 8,
            "borderTopLeftRadius": 8,
            "justifyContent": "center",
            "width": 56,
          },
        ]
      }
      variant="tertiary"
    >
      <Text
        inverted={true}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Open Sans",
              "fontSize": 14,
              "fontWeight": "600",
              "lineHeight": 20,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 8,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
      >
        thu
      </Text>
      <Text
        inverted={true}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Open Sans",
              "fontSize": 20,
              "fontWeight": "700",
              "lineHeight": 24,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
      >
        19
      </Text>
      <Text
        inverted={true}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Open Sans",
              "fontSize": 14,
              "fontWeight": "600",
              "lineHeight": 20,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
      >
        dec
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#41414A",
              "fontFamily": "Open Sans",
              "fontSize": 16,
              "fontWeight": "600",
              "lineHeight": 20,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 8,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
      >
        Yoga Class
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#6B6B78",
              "fontFamily": "Open Sans",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 8,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
      >
        Gympass
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#41414A",
              "fontFamily": "Open Sans",
              "fontSize": 12,
              "fontWeight": "400",
              "lineHeight": 16,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
      >
        19 am
      </Text>
    </View>
  </View>
</View>
`;
